Upgrade doc fixes (#6809)
solo-io/gloo: docs/content/operations/upgrading/v1.11.md

Helm maintainers, given the risk associated with changing CRDs. Given this limitation, you must apply new CRDs to the cluster before upgrading.
{{% /notice %}}

## Helm changes

> In Gloo Edge v1.11.20 / Gloo Edge Enterprise v1.11.27 and above, the default Gateways, ExtAuth Upstreams, and RateLimit Upstreams are installed via a Job during the Helm install/upgrade process, instead of included directly in the Helm chart. Thus after upgrading, those resources will no longer have Helm annotations (e.g. `meta.helm.sh/release-name`, `meta.helm.sh/release-namespace`) on them.

Reviewer: ```suggestion
In Gloo Edge v1.11.20 / Gloo Edge Enterprise v1.11.27 and later, the default Gateways, ExtAuth Upstreams, and RateLimit Upstreams are installed via a Job during the Helm installation and upgrade process, instead of being included directly in the Helm chart. After you upgrade to these versions, those resources no longer have Helm annotations (e.g. `meta.helm.sh/release-name`, `meta.helm.sh/release-namespace`).
```

{{% notice note %}}

> If you need to roll back to a previous Gloo Edge version, you will likely need to manually add back the Helm release annotations before performing the rollback.

Reviewer: Should we provide an example of how to add back these annotations, or nah?

```suggestion
If you need to roll back to an earlier Gloo Edge version, you must manually add the Helm release annotations back to these resources before performing the rollback.
```

Author: ha yeah i tried having it with the regular text as well as separately in a note. i can move it back to the paragraph

{{% /notice %}}

{{% notice warning %}}

> We strongly recommend that you do not manually edit (e.g. `kubectl apply` or `kubectl edit`) the Gloo Edge custom resources (e.g. Gateways, Upstreams) that are installed with the Helm chart, as this will likely cause errors on future upgrades.

Reviewer: ```suggestion
Do not manually edit (e.g. `kubectl apply` or `kubectl edit`) the Gloo Edge custom resources (e.g. Gateways, Upstreams) that are installed with the Helm chart, as this might cause errors in future upgrades.
```

<br/><br/>
If a resource has already been manually edited, then before starting the upgrade, make sure the resource's `metadata.annotations.kubectl.kubernetes.io/last-applied-configuration` annotation does not contain a `resourceVersion` field, as the existence of this field can cause conflicts on upgrade.
<br/><br/>

> If the field exists, it can be removed in either of the following ways:

Reviewer: ```suggestion
If the field exists, you can remove it in either of the following ways:
```

   kubectl delete crd graphqlapis.graphql.gloo.solo.io
   ```

> 4. Apply the new and updated CRDs (replace the version with the specific patch version that you are upgrading to):

Reviewer: ```suggestion
4. Apply the new and updated CRDs. Replace the version with the specific patch version that you are upgrading to.
```

   {{< tabs >}}
   {{% tab name="Gloo Edge" %}}
   ```sh
   helm repo update
   helm pull gloo/gloo --version 1.11.0 --untar
   kubectl apply -f gloo/crds
   ```
   {{% /tab %}}
   {{% tab name="Gloo Edge Enterprise" %}}
   ```sh
   helm repo update
   helm pull glooe/gloo-ee --version 1.11.0 --untar
   kubectl apply -f gloo-ee/charts/gloo/crds
   # If Gloo Federation is enabled
   kubectl apply -f gloo-ee/charts/gloo-fed/crds
   ```
   {{% /tab %}}
   {{< /tabs >}}

5. Continue to [upgrade the Gloo Edge server components]({{% versioned_link_path fromRoot="/operations/upgrading/upgrade_steps/#step-4-upgrade-gloo-edge" %}}) via Helm.

> 6. Gloo Federation only: Due to the addition of the GraphQLApi CRD in v1.11, the Gloo Federation ClusterRole needs to be updated with new permissions on each of the registered clusters. To do so, follow the instructions to [re-register]({{< versioned_link_path fromRoot="/guides/gloo_federation/cluster_registration/" >}}) each of your clusters.

Reviewer: ```suggestion
6. Gloo Federation only: Due to the addition of the GraphQLApi CRD in v1.11, the Gloo Federation ClusterRole in each registered cluster must be updated with new permissions. To do so, follow the instructions to [re-register]({{< versioned_link_path fromRoot="/guides/gloo_federation/cluster_registration/" >}}) each of your clusters.
```
